aws_batch/pipeline.py

```diff
@@ -387,27 +427,49 @@ def filter(self, record):
         self.last = record.relativeCreated
         return True
 
+def run_and_log(logparams, func_name, *args):
+    logger = logging.getLogger()
+    logger.info("========== %s ==========" % logparams.get("title"))
+    # produce the output
+    func_return = func_name(*args)
+    if func_return == 1:
+        logger.info("output exists, lazy run")
+    else:
+        logger.info("uploaded output")
+    # count records
+    if logparams["count_reads"]:
+        records_before = count_reads(logparams["before_file_name"], logparams["before_file_type"])
+        records_after = count_reads(logparams["after_file_name"], logparams["after_file_type"])
+        percent_removed = (100.0 * (records_before - records_after)) / records_before
+        if "paired" in logparams["before_file_type"] and "paired" in logparams["after_file_type"]:
+            record_description = "read-pairs"
+        else:
+            record_description = "reads"
+        logger.info("%s %% of %s dropped out, %s %s remaining" % (str(percent_removed), record_description,
+                                                                  str(records_after), record_description))
+    # copy log file
+    execute_command("aws s3 cp %s/%s %s/;" % (logparams["result_dir"], LOGS_OUT, logparams["sample_s3_output_path"]))
```

Comment:
Isn't this a lot cleaner? 
Also I would suggest create a different log.txt for each batch job

so it can be like log.txt.{job_id} 

you can access batch job id with 
os.environ['AWS_BATCH_JOB_ID']

Reply:
instead of hardcode the log filename
do 

logger.handlers[0].baseFilename  instead.

Reply:
ok